File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        res, val, prev = 0, 0, 0
        for n in nums:
            if n == 0:
                res = max(res, prev + val)
                prev = val
                val = 0
            else:
                val += n
        res = max(res, prev + val)
        return min(res, len(nums) - 1)

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubarray = function(nums) {
    let [res, prev, val] = [0, 0, 0];
    for (const n of nums) {
        if (n === 0) {
            res = Math.max(res, prev + val);
            prev = val;
            val = 0
        } else {
            val += n;
        }
    }
    res = Math.max(res, prev + val);
    return Math.min(res, nums.length - 1);
};